packages/core/adder/options.ts

```diff
@@ -137,12 +137,27 @@ export function prepareAndParseCliOptions<Args extends OptionDefinition>(adderDe
     const options = program.opts();
 
     if (multipleAdders) {
-        options.adder = program.args;
+        const selectedAdderIds: string[] = program.args ?? [];
+        validateAddersOrExit<Args>(adderDetails, selectedAdderIds);
+
+        options.adder = selectedAdderIds;
     }
 
     return options;
 }
 
+function validateAddersOrExit<Args extends OptionDefinition>(adderDetails: AdderDetails<Args>[], selectedAdderIds: string[]) {
```

Comment:
I think it could just be called `validateAdders`. I think "validate" often implies that it will throw an error if an issue is encountered
```suggestion
        const selectedAdderIds: string[] = program.args ?? [];
        validateAdders<Args>(adderDetails, selectedAdderIds);

        options.adder = selectedAdderIds;
    }

    return options;
}

function validateAdders<Args extends OptionDefinition>(adderDetails: AdderDetails<Args>[], selectedAdderIds: string[]) {
```